### ISSUE TITLE
Custom exports

### ISSUE THREAD

**Issue by beyer-johannes** [nuxt/module-builder#171]
Short description:
My nuxt module does export following in module.ts: export const car = 'something';
From an other module or package I want to import the car like this: import {car} from 'myCarModule'

Long description:
Seperating my project into multiple modules, each module contain mainly two things:
- Nuxt relevant things like plugins, middleware, components, server routes etc.
- Nuxt independed things like functions, types etc.

Example:
I have a module which contains everyting to CRUD a car. The database schema, api route, vue components.
Now I just want to have access from an other package like a cli tool to the database schema, because the tool seed the database with test cars.

This is actually not possible.

In other words, if I want to have access to not nuxt things, I need to create a seperate npm package just to store and share nuxt unrelevant things and create a nuxt module on top which use it.

A simple solution whould be, to export all types and modules, which are exported from the module.ts

And no, the virutal imports does not work for this case, because other packages does not know about '#myCarModule'.

**Comment by johannschopplich** [comment]
Good point, I had the same requirement. As of now, you can solve that with a custom `build.config.ts`. See an example in my [`nuxt-prepare` module](https://github.com/johannschopplich/nuxt-prepare/blob/main/build.config.ts) – the package provides a custom `/config` sub-path export, which I need in non-Nuxt environments.

The given exports in `build.config.ts` will be consumed from the module builder. I guess a simpler configuration for this use-case or documentation comes in handy nonetheless.

**Comment by beyer-johannes** [comment]
Thanks a lot @johannschopplich! My Intelij has problem with the types but the code basicly works.

**Comment by johannschopplich** [comment]
@beyer-johannes Make sure to add the [`typesVersions`](https://github.com/johannschopplich/nuxt-prepare/blob/c44e82cc24e862212be4ebb49229399029d5e42d/package.json#L35). If the user doesn't use bundler module resolution, type issue may appear, which seems like the case for you.

**Comment by beyer-johannes** [comment]
Mhm I copied the section from your nuxt-prepare project but still does not work.

 ```
 "typesVersions": {
    "*": {
      "*": [
        "*",
        "dist/*",
        "dist/*.d.ts"
      ]
    }
  },
```
But I need to say, I do not realy understand whats happen behind the magic :D
